include/seastar/coroutine/exception.hh

```diff
@@ -68,7 +68,13 @@ namespace coroutine {
 /// ```
 struct exception {
     std::exception_ptr eptr;
-    explicit exception(std::exception_ptr eptr) : eptr(std::move(eptr)) {}
+
+    explicit exception(std::exception_ptr&& eptr) noexcept : eptr(std::move(eptr)) {}
+
+    template <typename E>
+    requires std::derived_from<std::remove_reference_t<E>, std::exception>
+    explicit exception(E&& e) noexcept : eptr(std::make_exception_ptr(std::forward<E>(e))) {
```

Comment:
How about: `explicit exception(std::derived_from<std::exception> auto&& e)` to reduce boilerplate.

Reply:
I'll try that. If it works, I'll push v4.

Reply:
sone in v4.  thanks for your suggestion